src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,106 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public IEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties).GetAwaiter().GetResult();
+
+            int tableCompletionCount = 0;
+
+            using (var frames = pDataSet.GetFrames())
+            {
+                while (frames.MoveNext())
+                {
+                    var frame = frames.Current;
+
+                    switch (frame.FrameType)
+                    {
+                        case FrameType.TableFragment:
+                            {
+                                var content = frame as ProgressiveDataSetDataTableFragmentFrame;
+                                object[] row;
+                                while(GetNextRow(content, out row))
+                                {
+                                    yield return row;
```

Comment:
This could lead to some mistakes, since the "row" is really the same row object every time.  If someone, for example, wrote ExecuteStreamableQuery().ToList()... it's now a list of 100 copies _of the same single row_.

We should return a new row object for each row.  It's only a small number of bytes each time, so it's not worth the performance to leave that strange behavior in the code.

Reply:
Oh! You didn't, nevermind. I retract my own statement. This is your helper method, not the weird existing one.